### PR TITLE
Make prediction async

### PR DESCRIPTION
**Pull request by davidotte**
### Change Description

*Try to be precise. You can additionally add comments to your PR, this might help the reviewer a lot.*

This PR involves the client side for making predictions async. Please also check the corresponding [Server PR](https://github.com/automl/tabpfn-server/pull/92). On the client side, we only need updates in client.py. We now send streaming requests and basically just loop over all messages we receive. Currently, if a duration estimation is received, I print the estimated duration and then show a progress bar that runs for this estimated duration. The question is now if we want to keep this initial print of estimated duration and if we might also want to add a message that is printed if the prediction is taking longer than expected.

**If you used new dependencies: Did you add them to `requirements.txt`?**

Yes. Should I also add version numbers?

**Who did you ping on Mattermost to review your PR? Please ping that person again whenever you are ready for another review.**

@noahho 

## Breaking changes

If you made any breaking changes, please update the version number.
Breaking changes are totally fine, we just need to make sure to keep the users informed and the server in sync.

**Does this PR break the API? If so, what is the corresponding server commit?**

Yes, PR mentioned above

**Does this PR break the user interface? If so, why?**

---
*Please do not mark comments/conversations as resolved unless you are the assigned reviewer. This helps maintain clarity during the review process.*

### CODE REVIEW

File: tabpfn_client/client.py [hunk 1]
```diff
@@ -258,14 +271,52 @@ def predict(
...
+                                progress_bar.n = progress_bar.total
+                                progress_bar.refresh()
+                                progress_bar.close()
+                        elif data["event"] == "error":
+                            if data["error_class"] == "GCPOverloaded":
+                                raise GCPOverloaded(data["detail"])
+                            elif data["error_class"] == "ValueError":
+                                raise ValueError(data["detail"])
+                            else:
+                                raise RuntimeError(data["detail"])
```
Comment (noahho) [inline tabpfn_client/client.py:318]: Add "error_class"

**Review by noahho — APPROVED** [review]
Looks good to me

File: tabpfn_client/client.py (hunk 1 above)
Comment (davidotte) [reply tabpfn_client/client.py:318]: What do you mean, add it in the RuntimeError message?

Comment (noahho) [reply tabpfn_client/client.py:320]: raise RuntimeError(data["error_class"] + data["detail"])

**Pull request merged**